src/engraving/libmscore/check.cpp

```diff
@@ -268,15 +270,23 @@ void Measure::fillGap(const Fraction& pos, const Fraction& len, track_idx_t trac
          len.numerator(), len.denominator(),
          stretch.numerator(), stretch.denominator(),
          track);
-    TDuration d;
-    d.setVal(len.ticks());
-    if (d.isValid()) {
+    std::vector<TDuration> durationList;
+    if (useGapRests) { // fill this gap with a single rest
+        TDuration d;
+        d.setVal(len.ticks());
```

Comment:
OK, thinking about https://github.com/musescore/MuseScore/issues/16841 - what does this do if there is no single duration that can represent the length?  The bug in question was a case of a half rest plus an eighth rest being deleted.  That's a gap you can't fill with a single rest.  I'd think you still need to use the duration list thing even for gap rests, but - I know almost nothing about gap rests, maybe they're special in that way?

Reply:
I think this code may have been a kludge.

I've always considered using the single rest when necessitated by pasting to be wrong.  The generated rest(s) should always be the same as if a partial measure (such as a pickup measure) were generated.  The code to generate rests for a partial measure should used whenever a partial measure of rests are generated at the beginning of a measure.